File: melt.py
import pandas as pd
from pandas.core.reshape import melt
import sys
import argparse

# setup argument parsing
parser = argparse.ArgumentParser(description='Melt data.')
parser.add_argument('--ids', help='comma-separated list of column names', nargs='+')
args = parser.parse_args()

# read csv from stdin into a dataframe
df = pd.read_csv(sys.stdin, low_memory=False)

df = melt(df, id_vars=args.ids)

# output dataframe to stdout
df.to_csv(sys.stdout, index=False)

File: melt.R
#! /usr/local/bin/Rscript --vanilla --default-packages=reshape,optparse

# load libraries
suppressPackageStartupMessages(library(argparse))
suppressPackageStartupMessages(library(reshape))

# setup parser
parser <- ArgumentParser()
parser$add_argument('--ids')
parser$add_argument('--columns')
args <- parser$parse_args()

# read data from stdin
d <- read.csv(file("stdin"), strip.white=TRUE)

# melt data
d <- melt(d, id=unlist(strsplit(args$ids, split=',')))

# rename data
names(d) <- unlist(strsplit(args$columns, split=','))

# write data to stdout
write.csv(d, stdout(), row.names=FALSE)